install/kubernetes/cilium/Chart.yaml

```diff
@@ -143,3 +143,11 @@ annotations:
       displayName: Cilium Pod IP Pool
       description: |
         CiliumPodIPPool defines an IP pool that can be used for pooled IPAM (i.e. the multi-pool IPAM mode).
+    - kind: CiliumWorldCIDRSet
```

Comment:
nit: we are having ciliumcidrgroups (i.e. groups vs sets), not sure if you are thinking about it and intentionally use `groups` vs `sets`.

Reply:
@squeed Should we rename to CiliumWorldCIDRGroups?

Reply:
Ordinarily I would totally agree - I _hate_ API inconsistencies. In this case, however, I would be afraid that CIDRGroup and WorldCIDRGroup might be thought of as two almost-identical mechanisms that differ only in scope. For example, CiliumNetworkPolicy and CiliumClusterwideNetworkPolicy.

However, CiliumCIDRGroup affects policy, whereas CiliumWorldCIDRSet configures routing. I wouldn't want to confuse the two.

If we can come up with a better name for the type, I would love to. CiliumCIDRRoute? CiliumStaticRoute?

Reply:
How about going in a different direction: could we make CiliumWorldCIDR{Set,Group} expose a [CiliumCIDRGroups](https://github.com/cilium/cilium/blob/87ac446b902cf0f49177f87cc1cfc96a250eaded/pkg/policy/api/cidr.go#L49) reference instead of raw CIDRs? Then the only difference that CiliumWorldCIDR{Set,Group} would have is to provide routing configuration API. This makes the CRDs layer on each other such that the link between the two CRDs are explicit and then the naming question becomes much simpler.

For example,

```
type CiliumWorldCIDRSetSpec struct {

	// Encapsulate is whether or not the CIDRs in this set
	// should be encapsulated.
	//
	// Currently, only false is supported.
	//
	// +kubebuilder:validation:Required
	Encapsulate *bool `json:"encapsulate"`

	// CIDRGroupRef is a list of CiliumCIDRGroup references.
	// Note: Currently only IPv4 is supported.
	//
	// +kubebuilder:validation:Required
	CIDRGroupRef [] `json:"cidrGroupRef"`
}
```

Reply:
Hmm, seems reasonable. @pippolo84, is this a gross misuse of CIDRGroup?

That said, does the extra redirection gain us anything? Just shuffling around bags of CIDRs.

Reply:
I gains us better UX because the user would in theory only need to manage CIDRs in their CIDRGroups. The World CRD just piggybacks off of that. Interoperability, I guess if you were to use one word. 

Because of this, we can mitigate concerns like

> Ordinarily I would totally agree - I hate API inconsistencies. In this case, however, I would be afraid that CIDRGroup and WorldCIDRGroup might be thought of as two almost-identical mechanisms that differ only in scope.

It becomes more obvious that the two CRDs are not "identical mechanisms" for the same thing. In fact, the World CRD is explicitly tied to the CIDRGroups, making is easier to say that they are different CRDs altogether.

Reply:
Hmm, yeah, that makes sense.

Reply:
@christarazi updated based on this suggestion.

Reply:
> Hmm, seems reasonable. @pippolo84, is this a gross misuse of CIDRGroup?
> 
> That said, does the extra redirection gain us anything? Just shuffling around bags of CIDRs.

From a UX perspective, I think the proposal is great. :+1: 

I'm just wondering how much the potentially increased usage of the CIliumCIDRGroup resource might impact the performance of the CNP/CCNP handling, given that we share [a single handler](https://github.com/cilium/cilium/blob/main/pkg/k8s/watchers/cilium_network_policy.go#L107-L204) for CNP/CCNP/CiliumCIDRGroup and we keep an [in-memory cache](https://github.com/cilium/cilium/blob/main/pkg/k8s/watchers/cilium_network_policy.go#L101) for all the CiliumCIDRGroups.
Do we have any kind of estimation on the numbers related to this feature?

Anyway, this should be solved rewriting the handling logic above with the statedb.

Reply:
Right now, I don't expect this to be particularly large. Probably around 10 or so CIDRs. In fact, relative to heavier-weight things like pods or services, this will be a rounding error.

Reply:
Thanks for implementing! Very clean!

Now I guess back to the hard question: should we rename to `CiliumWorldCIDRGroups` now that it references CIDR groups? :)

Reply:
Btw, small typo in the commit msg: mentions `cidrs` as the key in the CWCG, but now it's `cidrGroupRefs`.

Reply:
> should we rename to CiliumWorldCIDRGroups

No, since I don't want to somehow conflate these with CiliumCIDRGroups. If anything, I'd call it CiliumCIDRGroupRoute, but I *also* don't want to pretend that this does any routing at all.

I fully expect this type to be deprecated and removed in a few releases.